tests/template_tests/syntax_tests/test_basic.py

```diff
@@ -346,6 +346,61 @@ def test_ignores_strings_that_look_like_format_interpolation(self):
         output = self.engine.render_to_string("tpl-weird-percent")
         self.assertEqual(output, "% %s")
 
+    @setup({"template": "{{ class_var.in_template }} | {{ class_var.render_class }}"})
+    def test_access_class_not_initiated(self):
+        class MyClass(list):
+            # As of Python 3.9 list defines __class_getitem__ which makes it
+            # subscriptable.
+            in_template = "Example property"
+            do_not_call_in_templates = True
```

Comment:
What's `do_not_call_in_templates` for? :thinking:

Reply:
I removed a comment `#  prevent instantiation` here
The test using a callable does not have `do_not_call_in_templates = True` so the subtext suggestion isn't quite equivalent

Reply:
I thought that `do_not_call_in_templates` was a made up variable, but I just learned that is something with a meaning! :blush:  For others wondering: https://docs.djangoproject.com/en/5.1/ref/templates/api/#variables-and-lookups

Still, I don't understand how `do_not_call_in_templates` is determinant in one test or the other... If it is, perhaps we need two tests: one withe `do_not_call_in_templates` set to False and one with it set to True?

Reply:
With `do_not_call_in_templates` set to `False`, the template engine will instantiate the class and access its property. The test will not be affected by the fix.